src/components/va-select/va-select.tsx

```diff
@@ -0,0 +1,141 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Prop,
+  State,
+  h,
+} from '@stencil/core';
+import { getSlottedNodes } from '../../utils/utils';
+
+@Component({
+  tag: 'va-select',
+  styleUrl: 'va-select.css',
+  shadow: true,
+})
+export class VaSelect {
+  @Element() el!: any;
+
+  /**
+   * Whether or not this is a required field.
+   */
+  @Prop() required: boolean;
+
+  /**
+   * Text label for the field.
+   */
+  @Prop() label: string;
+
+  /**
+   * Name attribute for the select field.
+   */
+  @Prop() name: string;
+
+  /**
+   * Selected value (will get updated on select).
+   */
+  @Prop({ reflect: true, mutable: true }) value: string;
+
+  /**
+   * Error message to display. When defined, this indicates an error.
+   */
+  @Prop() error: string;
+
+  /**
+   * Optional text that is displayed to screen readers only.
+   */
+  @Prop() ariaLiveRegionText: string;
+
+  /**
+   * Whether or not to fire the analytics events
+   */
+  @Prop() enableAnalytics: boolean;
+
+  @Event() keyDown: EventEmitter;
+
+  @Event() select: EventEmitter;
+
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  @State() options: Array<Node>;
+
+  private handleKeyDown(e: Event) {
+    this.keyDown.emit();
+  }
+
+  private handleChange(e: Event) {
+    const target: HTMLSelectElement = e.target as HTMLSelectElement;
+    this.value = target.value;
+
+    if (this.enableAnalytics) {
+      const detail = {
+        componentName: 'Select',
+        action: 'change',
+        details: {
+          label: this.label,
+          selectLabel: this.value,
+        },
+      };
+      this.componentLibraryAnalytics.emit(detail);
+    }
+    this.select.emit({ value: this.value });
+  }
+
+  componentDidRender() {
+    const { value } = this;
+
+    if (!this.options)
+      this.options = getSlottedNodes(this.el, 'option').map(
+        (node: HTMLOptionElement) => {
+          return (
+            <option value={node.value} selected={value === node.value}>
+              {node.text}
+            </option>
+          );
+        },
+      );
+  }
```

Comment:
So if we dynamically update the options (slotted children), they'll get rendered to the light DOM _and_ be reflected in the state variable, is that right? Is there any way that this might cause a disagreement between what's rendered to the DOM and what the state variable holds?

Reply:
The previous implementation (lifecycle method + conditional assignment) did _not_ render any updated nodes from the light DOM. However, the most recent handful of commits have fixed this by removing the conditional assignment and having the function be a handler for the `slotchange` event.